src/test/java/com/wikia/webdriver/testcases/auth/SignupTests.java

```diff
@@ -0,0 +1,288 @@
+    assertTrue(discussionPage.waitForPageReload().isStringInURL(PostsListPage.FULL_PATH),
+      "User should be redirected to discussion post list view upon sign up");
+  }
+
+  @Test(groups = DESKTOP)
+  public void passwordTogglerChangesPasswordVisibilityDesktop() {
+    RegisterPage signUpPage = openSignUpModalOnDesktop();
+    signUpPage.typePassword(existingUser.getPassword());
+
+    assertTrue(signUpPage.isPasswordMasked(), "password should be masked");
+    signUpPage.togglePasswordVisibility();
+    assertFalse(signUpPage.isPasswordMasked(), "password should be readable");
+  }
+
+  @Test(groups = MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  public void passwordTogglerChangesPasswordVisibilityMobile() {
+    RegisterPage signUpPage = navigateToSignUpOnMobile();
+    signUpPage.typePassword("");
+
+    assertTrue(signUpPage.isPasswordMasked(), "password should be masked");
+    signUpPage.togglePasswordVisibility();
+    assertFalse(signUpPage.isPasswordMasked(), "password should be readable");
+  }
+
+  /**
+   * HELPER METHODS
+   */
+
+  private SignUpUser createNewUser() {
+    return new SignUpUser(
+      String.format("QA%s", Instant.now().getEpochSecond()),
+      getEmailAlias(userWithEmail.getEmail()),
+      String.format("pass_%s", Instant.now().getEpochSecond()),
+      LocalDate.of(1993, 3, 19)
+    );
+  }
+
+  private SignUpUser createUserWithExistingEmail() {
+    return new SignUpUser(
+      String.format("QA%s", Instant.now().getEpochSecond()),
+      userWithEmail.getEmail(),
+      String.format("pass_%s", Instant.now().getEpochSecond()),
+      LocalDate.of(1993, 3, 19)
+    );
+  }
+
+  private SignUpUser createUserWithExistingUsername() {
+    return new SignUpUser(
+      existingUser.getUserName(),
+      getEmailAlias(userWithEmail.getEmail()),
+      String.format("pass_%s", Instant.now().getEpochSecond()),
+      LocalDate.of(1993, 3, 19)
+    );
+  }
+
+  private SignUpUser createUserWithPasswordMatchingUsername() {
+    String username = String.format("QA%s", Instant.now().getEpochSecond());
+    return new SignUpUser(
+      username,
+      getEmailAlias(userWithEmail.getEmail()),
+      username,
+      LocalDate.of(1993, 3, 19)
+    );
+  }
+
+  private SignUpUser createTooYoungUser() {
+    return new SignUpUser(
+      String.format("QA%s", Instant.now().getEpochSecond()),
+      getEmailAlias(userWithEmail.getEmail()),
+      String.format("pass_%s", Instant.now().getEpochSecond()),
+      LocalDate.now().minusYears(11)
+    );
+  }
+
+  private SignUpUser createNewUserWithSpecialCharacters() {
+    return new SignUpUser(
+      String.format("ユーザー%s", Instant.now().getEpochSecond()),
+      getEmailAlias(userWithEmail.getEmail()),
+      String.format("ユーザザー_%s", Instant.now().getEpochSecond()),
+      LocalDate.of(1993, 3, 19)
+    );
+  }
+
+  private String getEmailAlias(String email) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "getEmailAlias" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)